c-scape/src/lib.rs

```diff
@@ -2423,10 +2424,69 @@ unsafe extern "C" fn nanosleep(
 
 // exec
 
+unsafe fn null_terminated_array<'a>(list: *const *const c_char) -> Vec<&'a CStr> {
+    let mut len = 0;
+    while !(*list.wrapping_add(len)).is_null() {
+        len += 1;
+    }
```

Comment:
Can this loop be replaced by a call to `strlen`?

Reply:
i dont think that would work, `strlen` counts the number of non zero bytes, not non-null pointers, which are bigger, so you would get the wrong result.

Reply:
Ah, oops, right :-).